### PR TITLE
[TIMOB-24206] iOS: Fix image leaks on JSCore, use JSCore by default inside iOS-core as well

### PR DESCRIPTION
**Pull request by hansemannn**
**JIRA:** https://jira.appcelerator.org/browse/TIMOB-24206

### CODE REVIEW

**Review by vijaysingh-axway — CHANGES_REQUESTED** [review]
On iOS 11, it is crashing after selecting 30 pics/approx 3 mints.

**Comment by hansemannn** [comment]
@vijaysingh-axway PR updated with the regards to the discussion on Teams. We have to extremely sensitive tests to ensure the JSGarbageCollect plays nice.

File: iphone/Classes/TiUIView.m [hunk 1]
```diff
@@ -605,6 +605,7 @@ - (void)setBackgroundImage_:(id)image
   if (backgroundRepeat) {
     [self renderRepeatedBackground:bgImage];
   } else {
+    RELEASE_TO_NIL([self backgroundImageLayer].contents);
```
Comment (vijaysingh-axway) [inline iphone/Classes/TiUIView.m]: This line is causing crash. If I remove this line, JSGarbageCollect is working nice. I'll do more testing.

**Review by vijaysingh-axway — APPROVED** [review]
CR passed.

**Pull request merged**